experimental/go/internalgsdk.go

```diff
@@ -0,0 +1,262 @@
+package gsdk
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+)
+
+type internalGsdk struct {
+	currentGameState GameState
+	// transitionToActiveEvent is used to block the ReadyForPlayers() method
+	transitionToActiveEvent       chan interface{}
+	transitionToActiveEventClosed bool
+	// signalHeartbeatEvent is used to signal the heartbeat loop to send a heartbeat because of a state change
+	signalHeartbeatEvent          chan interface{}
+	configuration                 *GsdkConfig
+	configMap                     map[string]string
+	healthCallback                func() bool
+	maintenanceCallback           func(time.Time)
+	shutdownCallback              func()
+	connectedPlayers              []ConnectedPlayer
+	initialPlayers                []string
+	cachedScheduledMaintenanceUtc time.Time
+	startInternalRan              bool
+	// startInternalMutex is used to ensure that the startCore() method is only called once
+	startInternalMutex *sync.Mutex
+	debugLogs          bool
+}
+
+// newInternalGsdk creates a new instance of the internalGsdk struct
+func newInternalGsdk() *internalGsdk {
+	return &internalGsdk{
+		currentGameState:              "",
+		transitionToActiveEvent:       make(chan interface{}),
+		transitionToActiveEventClosed: false,
+		signalHeartbeatEvent:          make(chan interface{}),
+		configuration:                 nil,
+		configMap:                     nil,
+		healthCallback:                nil,
+		maintenanceCallback:           nil,
+		shutdownCallback:              nil,
+		connectedPlayers:              nil,
+		initialPlayers:                nil,
+		startInternalRan:              false,
+		startInternalMutex:            &sync.Mutex{},
+	}
+}
+
+// startInternal initializes the core GSDK functionality
+func (i *internalGsdk) startInternal() {
+	i.startInternalMutex.Lock()
+	defer i.startInternalMutex.Unlock()
+	if i.startInternalRan {
+		return
+	}
+
+	i.currentGameState = GameStateInitializing
+	var err error
+	if i.configuration == nil {
+		i.configuration, err = i.getConfiguration()
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Just curious.  Will this work as expected given that this is called before the `initializeLogger` method on line 80?

Reply:
so I had to do this since we rely on the configuration values to initialize the logger. I'm `pretty sure (TM)` this will work and panic the program, it will just not have the configuration values (which we have failed to retrieve anyway)